docs/data/data-grid/column-menu/ReuseColumnMenuGrid.tsx

```diff
@@ -0,0 +1,55 @@
+import * as React from 'react';
+import Button from '@mui/material/Button';
+import IconFilter from '@mui/icons-material/FilterAlt';
+import {
+  DataGrid,
+  GridColumnMenuDefault,
+  GridColumnMenuProps,
+  gridColumnMenuSlots,
+  GridColumnMenuItemProps,
+} from '@mui/x-data-grid';
+import { useDemoData } from '@mui/x-data-grid-generator';
+
+function CustomFilterItem(props: GridColumnMenuItemProps) {
+  const { column, onClick } = props;
+  return (
+    <Button sx={{ m: 1 }} onClick={onClick} startIcon={<IconFilter />}>
+      Show {column.field} Filters
+    </Button>
+  );
+}
+
+function CustomColumnMenu(props: GridColumnMenuProps) {
+  return (
+    <GridColumnMenuDefault
+      {...props}
+      slots={{
+        ...gridColumnMenuSlots,
+        ColumnMenuFilterItem: {
+          // override component for Filter item
+          component: CustomFilterItem,
+          // change order
+          priority: 1,
```

Comment:
Yeah, it might be more closer to the `components` and `componentsProps` pattern we have but it differs in a way that for `componentsProps`, the props passed are actually the props for the component in question and are passed over to that component. But in this case it's determining only the order of the item, kind of a metadata not props. So I am not sure if it'll make sense to call it `slotsProps`, as the users may expect to receive these props inside the components, even if they pass a custom one? 🤔

Reply:
My main concern is consistency with `components` (soon to be renamed to `slots`) in the MUI ecosystem.
We should keep this signature everywhere to avoid confusion:
```
slots={{ SlotName: SlotComponent }}
```

> So I am not sure if it'll make sense to call it slotsProps, as the users may expect to receive these props inside the components

If passing the `priority` prop to the slot allows keeping the slot signature I mentioned above - I think it's okay.
Is it doable?

Reply:
Makes sense, I've updated the code to accept `slots` and `slotsProps`.
Another minor inconsistency is that for both `slots` and `slotsProps` we are using **PascalCase** instead of **camelCase** for `slotsProps`, I guess it's fine for now, what do you think?

Reply:
I've created an issue for props renaming: https://github.com/mui/mui-x/issues/6986
For now, I think we should stick to the current naming convention:
```tsx
<DataGrid
  components={{ Component: Component }}
  componentsProps={{ component: { prop: value } }}
/>
```

Reply:
We discussed it on Monday in the eXplore team
It's not clear if we should support both in v6 or just rename and drop the old nomenclature

So yes, let's start with the old one and rename / duplicate when we rename / duplicate the other slots